### DISCUSSION TITLE
tutorials/apollo-ios-swift-part1/03-add-schema-codegen

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#181] [course feedback]
# tutorials/apollo-ios-swift-part1/03-add-schema-codegen

https://www.apollographql.com/tutorials/apollo-ios-swift-part1/03-add-schema-codegen

**Comment by ruvalcaba18** [comment]
Hi Apollo team and community 👋,

I'm encountering some issues with code generation in my iOS project and wanted to share feedback and suggestions. Currently, I’m using this script in a Build Phase to automatically generate classes from my GraphQL schema:

# Derived Data Path where the source packages are checked out.
DERIVED_DATA_CANDIDATE="${BUILD_ROOT}"

while ! [ -d "${DERIVED_DATA_CANDIDATE}/SourcePackages" ]; do
  if [ "${DERIVED_DATA_CANDIDATE}" = / ]; then
    echo >&2 "error: Unable to locate SourcePackages directory from BUILD_ROOT: '${BUILD_ROOT}'"
    exit 1
  fi

  DERIVED_DATA_CANDIDATE="$(dirname "${DERIVED_DATA_CANDIDATE}")"
done

# Grab a reference to the directory where scripts are checked out
SCRIPT_PATH="${DERIVED_DATA_CANDIDATE}/SourcePackages/checkouts/apollo-ios/scripts"

if [ -z "${SCRIPT_PATH}" ]; then
    echo >&2 "error: Couldn't find the CLI script in your checked out SPM packages; make sure to add the framework to your project."
    exit 1
fi

cd "${SRCROOT}/${TARGET_NAME}"
"${SCRIPT_PATH}"/run-bundled-codegen.sh codegen:generate --target=swift --includes=./**/*.graphql --localSchemaFile="schema.json" --output GraphQL/API.swift


Issue
When using this script with Apollo iOS 1.15.3 in Xcode 16.1, I noticed that the file run-bundled-codegen.sh is no longer automatically downloaded when including Apollo through SPM. This causes the code generation process to fail.

Requests
Documentation or examples for embeddedInTarget: I’m trying to use apollo-ios-cli init with the --module-type embeddedInTarget option, but I haven’t found any clear examples demonstrating the complete integration process. It would also be helpful to see examples for other options like swiftPackageManager or other.
Simplify the process: A more direct command to automate class generation in iOS without relying on additional scripts would be extremely helpful. For example:
apollo-ios-cli generate --embeddedInTarget --output ./GraphQL/API.swift

Restore or improve run-bundled-codegen.sh support: If this script is no longer available by design, could you provide an updated recommendation for automating code generation with SPM?

Any guidance or suggestions would be greatly appreciated. I believe these enhancements would benefit many developers in the community. 😊

**Comment by calvincestari** [comment]
Hi @ruvalcaba18, thanks for the feedback.

> When using this script with Apollo iOS 1.15.3 in Xcode 16.1, I noticed that the file run-bundled-codegen.sh is no longer automatically downloaded when including Apollo through SPM. This causes the code generation process to fail.

> Restore or improve run-bundled-codegen.sh support: If this script is no longer available by design, could you provide an updated recommendation for automating code generation with SPM?

The build script you've referenced is from the legacy documentation (0.x versions). It shouldn't be mentioned in the tutorial or any of our current documentation for version 1.x. If you do find reference to it please let me know and we'll correct it.

The script was removed because we no longer recommend running Apollo's code generation as an Xcode build phase. This is already detailed in [step 3](https://www.apollographql.com/docs/ios/migrations/1.0#step-3-replace-the-code-generation-build-phase) of our [1.0 Migration Guide](https://www.apollographql.com/docs/ios/migrations/1.0).

> Documentation or examples for embeddedInTarget: I’m trying to use apollo-ios-cli init with the --module-type embeddedInTarget option, but I haven’t found any clear examples demonstrating the complete integration process. It would also be helpful to see examples for other options like swiftPackageManager or other

There are many different configurations of schema and operation output, too many to have examples for all. We provide a more detailed example for generating the output into a Swift package because that is the recommended approach for most users.

> Simplify the process: A more direct command to automate class generation in iOS without relying on additional scripts would be extremely helpful. For example:
apollo-ios-cli generate --embeddedInTarget --output ./GraphQL/API.swift

The `init` command is required because there are a handful of configuration options that are required to be set by the user. The `init` command therefore creates a configuration file for customization before attempting code generation.

**Comment by jimbo51235** [comment]
Under Create your Codegen Configuration, you write the following.  "Next we need to set up our codegen configuration file. To do this run the following command in a terminal opened to the root of your project directory."  Which one is the root directory?  RocketReserver?

**Reply by jimbo51235** [reply]
Or is the starter folder that contains the graphql folder, the RocketReserver folder and the RocketReserver.xcodeproj container file?

**Comment by jimbo51235** [comment]
The root directory what

**Reply by jimbo51235** [reply]
Taking a look at the final folder, I find the apollo-codegen-config.json file inside of it.  So I suppose the root directory is the starter folder.  Switching to it and running ls -a, I don't find the command line tool.

**Reply by jimbo51235** [reply]
I have right-click on the Install CLI command several times.

**Reply by jimbo51235** [reply]
*right-clicked on

**Reply by jimbo51235** [reply]
I mean, right-clicked on...

**Reply by jimbo51235** [reply]
I don't find the command line tool in the RocketReserver folder inside of the DerivedData folder.  There is no such a hidden folder named 'build' inside of the RocketReserver folder, either.  There is a non-hidden folder named 'Build' inside the RocketReserver folder.

**Reply by jimbo51235** [reply]
My Xcode version is not 14.3.  It's 16.4.

**Reply by jimbo51235** [reply]
I see.  The root of the problem comes from the Select the latest minor version section in Page #1 where you tell us to set minor version to 1.15.x.  If I download the latest version (1.23.0), all problems are gone.

**Comment by udayabhishek** [comment]
I have faced two minor issue while running commands, got it fixed by changing name

**Reply by udayabhishek** [reply]
./apollo-ios-cli init --schema-namespace RocketReserverAPI --module-type swiftPackageManager to

./apollo-ios-cli init --schema-namespace RocketReserverAPI --module-type swiftPackage
